Source/Core/Core/HW/EXI/EXI_DeviceEthernetBase.h

```diff
@@ -210,7 +194,100 @@ class CEXIETHERNET : public IEXIDevice
   void DMARead(u32 addr, u32 size) override;
   void DoState(PointerWrap& p) override;
 
-private:
+protected:
+  static constexpr std::size_t BBA_RECV_SIZE = 0x800;
+  static constexpr std::size_t BBA_NUM_PAGES = 0x10;
+  static constexpr std::size_t BBA_PAGE_SIZE = 0x100;
+  static constexpr std::size_t BBA_MEM_SIZE = BBA_NUM_PAGES * BBA_PAGE_SIZE;
+  static constexpr std::size_t BBA_TXFIFO_SIZE = 1518;
+
+  virtual bool Activate() = 0;
+  virtual bool IsActivated() const = 0;
+  virtual bool SendFrame(const u8* frame, u32 size) = 0;
+  virtual bool RecvInit() = 0;
+  virtual void RecvStart() = 0;
+  virtual void RecvStop() = 0;
+
+  inline u16 page_ptr(int const index) const
+  {
+    return ((u16)m_bba_mem.raw[index + 1] << 8) | m_bba_mem.raw[index];
+  }
+
+  inline u8* PtrFromPagePtr(int const index)
+  {
+    return &m_bba_mem.raw[page_ptr(index) << 8];
+  }
+
+  inline const u8* PtrFromPagePtr(int const index) const
+  {
+    return &m_bba_mem.raw[page_ptr(index) << 8];
+  }
+
+  bool IsMXCommand(u32 const data);
+  bool IsWriteCommand(u32 const data);
+  const char* GetRegisterName() const;
+  void MXHardReset();
+  void MXCommandHandler(u32 data, u32 size);
+  void DirectFIFOWrite(const u8* data, u32 size);
+  void SendFromDirectFIFO();
+  void SendFromPacketBuffer();
+  void SendComplete();
+  u8 HashIndex(const u8* dest_eth_addr) const;
+  bool RecvMACFilter() const;
+  void inc_rwp();
+  bool RecvHandlePacket();
+
+  union {
+    std::array<u8, BBA_MEM_SIZE> raw;
+#pragma pack(push, 1)
+    struct {
+      u8 ncra;        //0x00
```

Comment:
I think for new code we're trying to avoid using C bitfields directly. The BitField class can abstract this away.

Reply:
There also isn't really a reason to use a bitfield here, it just makes it more messy.

Reply:
I don't need the bitfield functionality, a uint8/uint16/uint32 also works for padding, the main reason I used bit fields was to avoid giving the paddings a NAME!
I really don't want padding0, padding1, padding2, padding3 to show up in my autocomplete

Reply:
with the existing code there are no padding fields, it's just a single array.

Reply:
The old code used one huge byte array to store the informations, my refactored version uses a nice struct with named members. There are still accesses on a "byte-level", so I am casting the pointer to a byte array. We need to keep the old "paddings", otherwise some members of the struct get misaligned. I did not want to give each padding a name, so I used bitfields instead. I do not need the bitfield functionality, just the no-name hack/feature!

Reply:
To be clear, do you want me to refactor those nameless bitfields to `uint8 padding0; uint16 padding1;` ?